common/install/docker

```diff
@@ -3,7 +3,7 @@
 set -e
 
 mkdir -p /etc/docker
-echo '{"storage-driver": "vfs"}' > /etc/docker/daemon.json
+echo '{"storage-driver": "overlay2"}' > /etc/docker/daemon.json
```

Comment:
Is this working with Windows and OSX?
The last time I tried it was not working.

Reply:
Not sure, but it would surprise me if not.

I'd expect that anonymous volumes bind mount a regular Linux file system from within the Docker VM. That location should be able to use `overlay2`.

Reply:
Have you used a volume last time you tried?

By default `/var/lib/docker` is part of the container, which uses `overlay2`, and afaik `overlay2` on `overlay2` is a nogo.

Reply:
I don't remember, I just know it had to be changed to work on Windows and OSX

Reply:
Ok. I can give it a try on Windows later, but I don't have an OSX machine. I'd expect if it works on Windows, it should be fine on OSX as well.

Reply:
Tested `overlay2` on Windows. My machine seems a bit on the lower end to work with Docker & VSCode (using a older Laptop currently), but I was able to start the devcontainer and Docker inside it with the `overlay2` driver. The Supervisor dev container was able to build and start containers.

Reply:
We need to add some sort of check here.
If you use a devcontainer that is created after this is added, and you don't adjust the JSON file, the docker service will fail.

Reply:
Yeah makes sense. I'll try to implement a check to see if `/var/lib/docker` is a location which can support an overlayfs.

Reply:
Implemented an error message: https://github.com/home-assistant/devcontainer/pull/5/files#diff-1e55b06733e95819c4e0eff6c1899ba2170f4965232067a795635c3a159bae2fR20-R25

E.g. when running Supervisor without the additional mount, it says now:

```
> Executing task: supervisor_run <

Run Supervisor
Docker storage directory /var/lib/docker is an overlayfs. The location needs
to be a bind mount (e.g. using a volume) to support the overlay2 backend.
The terminal process "/usr/bin/zsh '-c', 'supervisor_run'" terminated with exit code: 1.

Press any key to close the terminal.
```

Alternatively, we also could select the backend based on what FS `/var/lib/docker` is, essentially automatically enable overlayfs if possible... It might slow down adoption of overlayfs backend in devcontainers though. Thoughts?

Reply:
Can't we do that check in the install script and fallback to vfs if not supported?

Reply:
Afaict that is not possible since this is done at `docker build` time, where `/var/lib/docker` is not yet what it will be when VSCode starts the dev container.

What we could do though instead of creating a static `daemon.json` at install time, we could just choose between vfs or overlay2 dynamically at startup time.

I'll give this a try, along with a hint that overlayfs can be used for better performance.

Reply:
That would be preferred, I'm not too keep on "randomly" breaking this (We don't do tags/releases here)

Reply:
With https://github.com/home-assistant/devcontainer/pull/5/commits/faa2a36760a7b1902f51f8db36335cca3b1c3637 the start script now determines the storage driver at runtime. When using `devcontainer.json` without the mount, VFS is used along with a hint:

```
> Executing task: supervisor_run <

Run Supervisor
Starting docker.
Using "vfs" storage driver. Bind mount /var/lib/docker to use the faster overlay2 driver.
Waiting for docker to initialize...
Docker was initialized
```